lib/core/sync_job_runner.rb

```diff
@@ -94,19 +90,31 @@ def do_sync!
 
         Utility::Logger.debug("#{existing_ids.size} documents are present in index #{@index_name}.")
 
-        post_processing_engine = Core::Filtering::PostProcessEngine.new(@job.filtering)
-        Utility::Logger.info('Yielding documents...')
-        @connector_instance.yield_documents do |document|
-          document = add_ingest_metadata(document)
-          post_process_result = post_processing_engine.process(document)
-          if post_process_result.is_include?
-            @sink.ingest(document)
-            incoming_ids << document['id']
+        # We want to validate advanced filtering rules even if basic rules are disabled
+        if @connector_settings.any_filtering_feature_enabled?
+          Utility::Logger.info("Checking active filtering for sync job #{@job_id} for connector #{@connector_id}.")
+          validate_filtering(@job.filtering)
+          Utility::Logger.debug("Active filtering for sync job #{@job_id} for connector #{@connector_id} is valid.")
+        end
```

Comment:
Why move this block here? If there is any invalid rule, `validate_filtering` will raise an exception and stop the sync? So it makes sense to execute this before `ElasticConnectorActions.fetch_document_ids`?

Reply:
Good point! That was unnecessary to move it down.